sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1775,6 +1775,48 @@ class CastSuite extends CastSuiteBase {
     }
   }
 
+  test("SPARK-35112: Cast string to day-time interval") {
+    checkEvaluation(cast(Literal.create("0 0:0:0"), DayTimeIntervalType), 0L)
+    checkEvaluation(cast(Literal.create("INTERVAL '0 0:0:0' DAY TO SECOND"),
```

Comment:
Could you check `INTERVAL` in lower case + spaces:
```suggestion
    checkEvaluation(cast(Literal.create(" interval '0 0:0:0' Day TO second   "),
```

Reply:
@AngersZhuuuu Your force push reverted the changes I approved. Please, revert them back.

Reply:
> @AngersZhuuuu Your force push reverted the changes I approved. Please, revert them back.

Updated